### PR TITLE
G3 322 cross species mapping in geneweaver client

### PR DESCRIPTION
**Pull request by bergsalex**
Adds a command that orchestrates the process of calling the AON and Geneweaver APIs to get cross species geneset values as Ensembl Mouse Gene IDs.

### CODE REVIEW

**Review by gerring — CHANGES_REQUESTED** [review]
Are there enough tests for failure?

**Comment by bergsalex** [comment]
> Are there enough tests for failure?

Waiting on final input from stakeholders before finishing up the tests.

**Review by gerring — APPROVED** [review]
Dont assume that I now know how to use AON API from reading this but I don't see anything obviously wrong.

**Review by desais-jax — APPROVED** [review]

File: src/geneweaver/client/api/aon.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    ORTHOFINDER = "OrthoFinder"
+    ZFIN = "ZFIN"
+    ENSEMBL_COMPARA = "Ensembl Compara"
+    SONICPARANOID = "SonicParanoid"
+    OMA = "OMA"
+    XENBASE = "Xenbase"
+
+
+def ortholog_mapping(
+    identifiers: List[str], to_species: Species, algorithm_id: Optional[int] = None
```
Comment (obwalton) [inline src/geneweaver/client/api/aon.py:29]: Why not have a default algorithm? Or is that buried somewhere?

Comment (bergsalex) [reply src/geneweaver/client/api/aon.py:29]: Two reasons.

1) After discussing with the stakeholder, it's not clear what the default should be.
2) No algorithm (`algorithm_id=None`) represents a valid call of the API. It represents an API request which doesn't filter the response by algorithm ID, which functionally returns mappings from **all algorithms**.

**Review by obwalton — APPROVED** [review]

**Pull request merged**